src/hal/src/command/render_pass.rs

```diff
@@ -109,70 +120,70 @@ impl<'a, B: Backend> RenderSubpassCommon<'a, B> {
         T: IntoIterator,
         T::Item: Borrow<pso::Rect>,
     {
-        self.0.set_scissors(first_scissor, scissors)
+        self.cmb.borrow_mut().set_scissors(first_scissor, scissors)
     }
 
     ///
     pub fn set_stencil_reference(&mut self, faces: pso::Face, value: pso::StencilValue) {
-        self.0.set_stencil_reference(faces, value);
+        self.cmb.borrow_mut().set_stencil_reference(faces, value);
     }
 
     ///
     pub fn set_stencil_read_mask(&mut self, faces: pso::Face, value: pso::StencilValue) {
-        self.0.set_stencil_read_mask(faces, value);
+        self.cmb.borrow_mut().set_stencil_read_mask(faces, value);
     }
 
     ///
     pub fn set_stencil_write_mask(&mut self, faces: pso::Face, value: pso::StencilValue) {
-        self.0.set_stencil_write_mask(faces, value);
+        self.cmb.borrow_mut().set_stencil_write_mask(faces, value);
     }
 
     ///
     pub fn set_blend_constants(&mut self, cv: pso::ColorValue) {
-        self.0.set_blend_constants(cv)
+        self.cmb.borrow_mut().set_blend_constants(cv)
     }
 
     ///
     pub fn set_depth_bounds(&mut self, bounds: Range<f32>) {
-        self.0.set_depth_bounds(bounds)
+        self.cmb.borrow_mut().set_depth_bounds(bounds)
     }
 
     ///
     pub fn push_graphics_constants(&mut self, layout: &B::PipelineLayout, stages: pso::ShaderStageFlags, offset: u32, constants: &[u32]) {
-        self.0.push_graphics_constants(layout, stages, offset, constants);
+        self.cmb.borrow_mut().push_graphics_constants(layout, stages, offset, constants);
     }
 
     ///
     pub fn set_line_width(&mut self, width: f32) {
-        self.0.set_line_width(width);
+        self.cmb.borrow_mut().set_line_width(width);
     }
 
     ///
     pub fn set_depth_bias(&mut self, depth_bias: pso::DepthBias) {
-        self.0.set_depth_bias(depth_bias);
+        self.cmb.borrow_mut().set_depth_bias(depth_bias);
     }
 
     // TODO: pipeline barrier (postponed)
 
     ///
     pub fn begin_query(&mut self, query: query::Query<B>, flags: query::ControlFlags) {
-        self.0.begin_query(query, flags)
+        self.cmb.borrow_mut().begin_query(query, flags)
     }
 
     ///
     pub fn end_query(&mut self, query: query::Query<B>) {
-        self.0.end_query(query)
+        self.cmb.borrow_mut().end_query(query)
     }
 
     ///
     pub fn write_timestamp(&mut self, stage: pso::PipelineStage, query: query::Query<B>) {
-        self.0.write_timestamp(stage, query)
+        self.cmb.borrow_mut().write_timestamp(stage, query)
     }
 }
 
 /// An object that records commands into a command buffer inline, that is,
 /// without secondary command buffers.
-pub struct RenderPassInlineEncoder<'a, B: Backend, L: Level>(pub(crate) Option<RenderSubpassCommon<'a, B>>, PhantomData<L>)
```

Comment:
`RenderPassInlineEncoder` cannot be on secondary level.

Reply:
nicely spotted!